core/src/main/java/hudson/Util.java

```diff
@@ -1094,8 +1094,8 @@ public static String singleQuote(String s) {
     /**
      * Escapes HTML unsafe characters like &lt;, &amp; to the respective character entities.
      */
-    @Nonnull
-    public static String escape(@Nonnull String text) {
+    @Nullable
+    public static String escape(@CheckForNull String text) {
```

Comment:
The check for null should propagate, either the parameter is nullable in which case the return type is too (and nullable is the package default) or the return type should be check for null (which it shouldn’t imho)

Reply:
I am not sure what you are proposing, exactly.

The existing behavior, which we cannot compatibly change, is that the return value will be null iff the parameter is null. So, the parameter is marked `@CheckForNull` as a reminder inside the method that we do, in fact, need to check for null, and a note to callers that passing null is permitted; and the return value is `@Nullable` to emphasize to callers that a null check may or may not be needed, depending on what you know about the parameter.